homeassistant/components/acaia/entity.py

```diff
@@ -0,0 +1,58 @@
+"""Base class for acaia entities."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyacaia_async.acaiascale import AcaiaScale
+
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import AcaiaCoordinator
+
+
+@dataclass(kw_only=True, frozen=True)
+class AcaiaEntityDescription(EntityDescription):
+    """Description for acaia entities."""
+
+    available_fn: Callable[[AcaiaScale], bool] = lambda scale: scale.connected
```

Comment:
Why do we have this on entity level?

Reply:
Will be reused across entities

Reply:
But we don't need it in this PR so let's not add it

Reply:
I am using it for the buttons already. Sure, I could move it there, but leaving it here would make the next platform PRs independent from each other...

Reply:
nvm